compiler/x/codegen/FPTreeEvaluator.cpp

```diff
@@ -198,6 +198,173 @@ void OMR::X86::TreeEvaluator::insertPrecisionAdjustment(TR::Register      *reg,
+         {
+         // secondChild => xmm1
+         preferredChildInTargetRegister = 1;
+         }
+      else if (preferredChildFromMemory == -1 && secondChild->getOpCode().isLoadVar())
+         {
+         // secondChild => xmm3/m32/m64
+         preferredChildFromMemory = 1;
+         }
+      }
+
+   if (thirdChild->getReferenceCount() == 1)
+      {
+      if (preferredChildInTargetRegister == -1 && thirdChild->getRegister() != NULL)
+         {
+         // thirdChild => xmm1
+         preferredChildInTargetRegister = 2;
+         }
+      else if (preferredChildFromMemory == -1 && thirdChild->getOpCode().isLoadVar())
+         {
+         // thirdChild => xmm3/m32/64
+         preferredChildFromMemory = 2;
+         }
+      }
+
+   int opcodeVariant, source1Choice, source2Choice, source3Choice;
+
+   if (preferredChildFromMemory != -1 && preferredChildInTargetRegister != -1)
+      {
+      source1Choice = preferredChildInTargetRegister;
+      source2Choice = 3 - preferredChildFromMemory - preferredChildInTargetRegister;
+      source3Choice = preferredChildFromMemory;
+
+      if (preferredChildFromMemory == 2)
+         opcodeVariant = 3;
+      else if (preferredChildInTargetRegister == 2)
+         opcodeVariant = 5;
+      else
+         opcodeVariant = 1;
+      }
+   else if (preferredChildFromMemory != -1)
+      {
+      source3Choice = preferredChildFromMemory;
+
+      if (preferredChildFromMemory == 2)
+         {
+         opcodeVariant = 3;
+         source1Choice = 0;
+         source2Choice = 1;
+         }
+      else
+         {
+         opcodeVariant = 1;
+         source1Choice = 1 - preferredChildFromMemory;
+         source2Choice = 2;
+         }
+      }
+   else if (preferredChildInTargetRegister != -1)
+      {
+      source1Choice = preferredChildInTargetRegister;
+      if (preferredChildInTargetRegister == 2)
+         {
+         opcodeVariant = 4;
+         source2Choice = 0;
+         source3Choice = 1;
+         }
+      else
+         {
+         opcodeVariant = 0;
+         source2Choice = 2;
+         source3Choice = 1 - preferredChildInTargetRegister;
+         }
+      }
+   else
+      {
+      opcodeVariant = 0;
+      source1Choice = 0;
+      source2Choice = 2;
+      source3Choice = 1;
+      }
+
+   TR::ILOpCodes nodeOp = node->getOpCodeValue();
+
+   int ilVariant;
+
+   switch (nodeOp)
+      {
+      case TR::fmadd:
+         ilVariant = 0;
+         break;
+      case TR::fmsub:
+         ilVariant = 1;
+         break;
+      case TR::fnmadd:
+         ilVariant = 2;
+         break;
+      case TR::fnmsub:
+         ilVariant = 3;
+         break;
+      case TR::dmadd:
+         ilVariant = 4;
+         break;
+      case TR::dmsub:
+         ilVariant = 5;
+         break;
+      case TR::dnmadd:
+         ilVariant = 6;
+         break;
+      case TR::dnmsub:
+         ilVariant = 7;
+         break;
+      }
```

Comment:
needs a default case, and ilVariant should get a value in that case

Reply:
But the fmaEvaluator will be called only for the 8 fma opcodes, so what instructions should I generate for the default case then?